src/support/msw/helpers/data.ts

```diff
@@ -82,3 +87,44 @@ export const newShoppingListWithAggregate = (
     },
   ]
 }
+
+/**
+ *
+ * Shopping list item creation
+ *
+ */
+
+export const newShoppingListItem = (
+  attributes: RequestShoppingListItem,
+  listId: number
+) => {
+  const list = allShoppingLists.find(({ id }) => id === listId)
+
+  if (!list)
+    throw new Error(`No shopping list with ID ${listId} in the test data`)
+
+  const shoppingList = { ...list }
```

Comment:
Have to duplicate the shopping list or else it actually changes the test data in memory and maintains that state between tests, which took me a while to figure out.